openwisp_controller/subnet_division/filters.py

```diff
@@ -0,0 +1,78 @@
+from django.contrib import admin
+from django.db.models import Q
+from swapper import load_model
+
+from . import settings as app_settings
+
+SubnetDivisionIndex = load_model('subnet_division', 'SubnetDivisionIndex')
+Subnet = load_model('openwisp_ipam', 'Subnet')
+
+
+class InputFilter(admin.SimpleListFilter):
+    template = 'admin/input_filter.html'
+
+    def lookups(self, request, model_admin):
+        # Required to show the filter.
+        return [tuple()]
+
+    def choices(self, changelist):
+        # Grab only the "all" option.
+        all_choice = next(super().choices(changelist))
+        all_choice['query_parts'] = []
+        for key, value in changelist.get_filters_params().items():
+            if key != self.parameter_name:
+                all_choice['query_parts'].append((key, value))
+        yield all_choice
+
+
+class SubnetFilter(InputFilter):
+    parameter_name = 'subnet'
+    title = 'Subnet'
+
+    def queryset(self, request, queryset):
+        if self.value() is not None:
+            subnet_text = self.value()
+            master_subnet_key = (
+                'config__subnetdivisionindex__subnet__master_subnet__subnet'
+            )
+            return queryset.filter(
+                Q(**{master_subnet_key: subnet_text})
+                | Q(config__subnetdivisionindex__subnet__subnet=subnet_text)
+            ).distinct()
+
+
+class DeviceFilter(InputFilter):
+    """
+    Filters Subnet queryset for input device name
+    using SubnetDivisionIndex
+    """
+
+    parameter_name = 'device'
+    title = 'Device'
+
+    def queryset(self, request, queryset):
+        if self.value() is not None:
+            device_text = self.value()
+
+            return queryset.filter(
+                id__in=SubnetDivisionIndex.objects.filter(
+                    config__device__name=device_text
```

Comment:
I'd simply call `self.value()` here without storing it in a var to keep it shorter since it would be readable anyway